### ISSUE TITLE
Clang-Format AlignAfterOpenBracket: BlockIndent improperly breaks after function pointer name

### ISSUE THREAD

**Issue by jaskij** [llvm/llvm-project#58496]
When using Clang-Format with **C** it seems that function pointers are handled improperly. A break is inserted before the closing parenthesis after pointer name. It seems that the variable name is treated as an argument and shenanigans ensue.

Consider the following code - this is how clang-format formats it right now:

```c
int (*tps25750_write
)(void*    ctx,
  uint8_t  tps_address,
  uint8_t  reg_address,
  uint8_t  byte_count,
  uint8_t* buffer);
```

I would expect it to look like this, closer to a function declaration (all my indents are set to 4 spaces):

```c
int (*tps25750_write)(
    void*    ctx,
    uint8_t  tps_address,
    uint8_t  reg_address,
    uint8_t  byte_count,
    uint8_t* buffer
    );
```

Full ~~[`.clang-format`](https://gist.github.com/jaskij/e2646cb88477d47d42ab5862dd677a31/440b832497a281023313e0fbfd83c120a06b8a1d)~~. That file mistakenly contained options from Clang-Format 16. [Current file which actually works with Clan-Format 14](https://gist.github.com/jaskij/e2646cb88477d47d42ab5862dd677a31/1a11c342b48a5b9621510344089f1683d59558e4)

**Comment by gedare** [comment]
I proposed a fix at https://reviews.llvm.org/D137762. I could not reproduce your exact scenario, but I did create a simple test case that is similar.

**Comment by jaskij** [comment]
Looking at the comments and examples in the fix, this does look like what I'd want.

Do you want me to try and reduce my code to a minimal reproducible example for another test case?

**Comment by gedare** [comment]
> Looking at the comments and examples in the fix, this does look like what I'd want.
> 
> Do you want me to try and reduce my code to a minimal reproducible example for another test case?

No, I think I have the testing side figured out. There are a few corner cases that don't work quite right still. I'm hopeful to work it out though.

**Comment by owenca** [comment]
Fixed in b40e9dce0a67.
